### PR TITLE
Adding last update time to ScheduleDetail that gets returned by get s…

### PR DESCRIPTION
**Pull request by wyzhang**
Include last update time of the schedule in the ScheduleDetail that gets returned by REST API call

This is in preparation of switching ProfileMetadataMessage from direct local levelDB access to REST calls.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java]: style: code blocks in {} should be on a new line.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/Scheduler.java
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/Scheduler.java:140]: unless this is cheaper to get (1 I/O vs multiple I/O or something like that), it seems like it should be replaced.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java]: is this object stored in a serialized form anywhere or returned as part of some Http endpoint? If so, changing the variable name will be a backwards incompatible change.

It's safer to just change the method name

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java
Comment (wyzhang) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java]: fixed.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/Scheduler.java
Comment (wyzhang) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/Scheduler.java:140]: Yes. Plan to deprecate it and replace with the new getScheduleRecord. 

Added a TODO to clean up in a follow-up change

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java
Comment (wyzhang) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ProgramScheduleMeta.java]: This looks to be used internally and transiently. We convert it to ScheduleDetail that gets returned to client. So the renaming looks to be safe. 

Anyway, I am reverting this renaming change, just to be on the safeside. Will do the renaming in a follow-up change and take a much closer look there.

**Review by albertshau — APPROVED** [review]
In the future, use a branch name that is of the form:

feature/[jira]-[description]

or

bugfix/[jira]-[description]

This will result in the build system (https://builds.cask.co/browse/CDAP) automatically creating a branch and running the tests. We normally will link to the corresponding build in the PR comments, though you'll have to manually create the branch for this one.

LGTM if the build is green, please squash commits before merging.

**Pull request closed without merging**
